go/vt/vtgate/buffer/shard_buffer.go

```diff
@@ -561,6 +562,18 @@ func (sb *shardBuffer) stopBufferingLocked(reason stopReason, details string) {
 	go sb.drain(q, clientEntryError)
 }
 
+// parallelRangeIndex uses counter to return a unique idx value up to the
+// passed max and ok will be set to false if the counter exceeds the max
+func parallelRangeIndex(counter *atomic.Int64, max int) (idx int, ok bool) {
+	next := counter.Add(1)
+	if next-1 > int64(max) {
+		return -1, false
+	}
+	// if this is a 32-bit platform, max won't exceed the 32-bit integer limit
+	// so a cast from a too-large 64-bit int to a 32-bit int will never happen
+	return int(next) - 1, true
```

Comment:
Any reason not to use int64 so that the behavior is deterministic / platform independent and we don't need to cast?

Reply:
slice indexes are already "int" (would have to be casted later) and the issue is mitigated internally in the function, which I think is better than forcing the caller to deal with it